File: common/src/main/java/dev/number6/db/model/converter/EntityTotalConverter.java
package dev.number6.db.model.converter;

import dev.number6.db.model.EntityTotal;
import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBTypeConverter;
import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;

import java.lang.reflect.Type;
import java.util.Collection;

public class EntityTotalConverter implements DynamoDBTypeConverter<String, Collection<EntityTotal>> {

    private final Gson gson = new Gson();

    @Override
    public String convert(Collection<EntityTotal> totals) {
        return gson.toJson(totals);
    }

    @Override
    public Collection<EntityTotal> unconvert(String object) {
        Type type = new TypeToken<Collection<EntityTotal>>() {
        }.getType();
        return gson.fromJson(object, type);
    }
}

File: common/src/main/kotlin/dev/number6/db/model/converter/EntityTotalConverter.kt
package dev.number6.db.model.converter

import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBTypeConverter
import com.google.gson.Gson
import com.google.gson.reflect.TypeToken
import dev.number6.db.model.EntityTotal

class EntityTotalConverter : DynamoDBTypeConverter<String, Collection<EntityTotal>> {
    private val gson = Gson()
    override fun convert(totals: Collection<EntityTotal>): String {
        return gson.toJson(totals)
    }

    override fun unconvert(`object`: String): Collection<EntityTotal> {
        val type = object : TypeToken<Collection<EntityTotal>>() {}.type
        return gson.fromJson(`object`, type)
    }
}